cmd/server/pactasrv/analysis.go

```diff
@@ -341,3 +283,88 @@ func (s *Server) RunAnalysis(ctx context.Context, request api.RunAnalysisRequest
 
 	return api.RunAnalysis200JSONResponse{AnalysisId: string(result)}, nil
 }
+
+func (s *Server) analysisDoAuthzAndAuditLog(ctx context.Context, analysisID pacta.AnalysisID, action pacta.AuditLogAction) error {
+	actorInfo, err := s.getActorInfoOrFail(ctx)
+	if err != nil {
+		return err
+	}
+	analysis, err := s.DB.Analysis(s.DB.NoTxn(ctx), analysisID)
+	if err != nil {
+		if db.IsNotFound(err) {
+			return notFoundOrUnauthorized(actorInfo, action, pacta.AuditLogTargetType_Analysis, analysisID)
+		}
+		return oapierr.Internal("querying analysis for authz failed", zap.Error(err))
+	}
+	as := &authzStatus{
+		primaryTargetID:      string(analysisID),
+		primaryTargetType:    pacta.AuditLogTargetType_Analysis,
+		primaryTargetOwnerID: analysis.Owner.ID,
+		actorInfo:            actorInfo,
+		action:               action,
+	}
+	switch action {
+	case pacta.AuditLogAction_Update, pacta.AuditLogAction_Delete, pacta.AuditLogAction_ReadMetadata:
+		as.isAuthorized, as.authorizedAsActorType = allowIfAdminOrOwner(actorInfo, analysis.Owner.ID)
+	default:
+		return fmt.Errorf("unknown action %q for analysis authz", action)
+
+	}
+	return s.auditLogIfAuthorizedOrFail(ctx, as)
```

Comment:
nit: Same note as in the previous PR, where the `OrFail` seems redundant, doubly so with the `IfAuthorized`.

Reply:
Fixed (in a hellish merge :))